rustdoc: on mobile, make the sidebar full width and linewrap

lolbinarycat opened the pull request:
this is because the mobile sidebar cannot be resized, unlike on desktop.

<!-- homu-ignore:start -->
<!--
If this PR is related to an unstable feature or an otherwise tracked effort,
please link to the relevant tracking issue here. If you don't know of a related
tracking issue or there are none, feel free to ignore this.

This PR will get automatically assigned to a reviewer. In case you would like
a specific user to review your work, you can assign it to them by using

    r? <reviewer name>
-->
<!-- homu-ignore:end -->

lolbinarycat committed:
rustdoc: on mobile, make the sidebar full width and linewrap

this is because the mobile sidebar cannot be resized,
unlike on desktop.

rust-log-analyzer commented:
The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/14458241914/job/40545889273) [(plain)](https://github.com/rust-lang/rust/commit/94558c5c80f77630450488757663e8d36ccd9a36/checks/40545889273/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
.......... (130/137)
....FFF    (137/137)


/checkout/tests/rustdoc-gui/notable-trait.goml notable-trait... FAILED
[ERROR] `tests/rustdoc-gui/notable-trait.goml` line 249: expected 1 elements, found 0: for command `assert-count: ("//*[@class='tooltip popover']", 1)`

/checkout/tests/rustdoc-gui/pocket-menu.goml pocket-menu... FAILED
[ERROR] `tests/rustdoc-gui/pocket-menu.goml` line 73: "#settings-menu .popover" not found: for command `assert-css: ("#settings-menu .popover", {"display": "block"})`
[ERROR] `tests/rustdoc-gui/pocket-menu.goml` line 81: "#settings-menu a" not found: for command `click: "#settings-menu a"`

/checkout/tests/rustdoc-gui/sidebar-mobile.goml sidebar-mobile... FAILED
[ERROR] `tests/rustdoc-gui/sidebar-mobile.goml` line 37: The following errors happened (for selector `.sidebar`): [expected `-1000px` for key `left`, found `0px`]: for command `assert-css: (".sidebar", {"display": "block", "left": "-1000px"})`
[ERROR] `tests/rustdoc-gui/sidebar-mobile.goml` line 41: The following errors happened (for selector `.sidebar`): [expected `0px` for key `left`, found `-1000px`]: for command `assert-css: (".sidebar", {"left": "0px"})`
[ERROR] `tests/rustdoc-gui/utils.goml` line 10
    from `tests/rustdoc-gui/sidebar-mobile.goml` line 66
    from `tests/rustdoc-gui/sidebar-mobile.goml` line 78: Error: The CSS selector "#settings" was not found: for command `wait-for: "#settings"`

Error: ()
Build completed unsuccessfully in 0:05:04
  local time: Tue Apr 15 01:02:00 UTC 2025
  network time: Tue, 15 Apr 2025 01:02:00 GMT
```

</details>

lolbinarycat commented:
@GuillaumeGomez I'm really not sure how changing the size of these elements could cause these test failures, got a clue?

GuillaumeGomez commented:
Please add a screenshot. Also, what's the reasoning behind this change? What would it improve?

lolbinarycat commented:
Motivation is how much stuff gets needlessly truncated:

![before screenshot](https://github.com/user-attachments/assets/fdc016f1-7294-456a-9d66-71401a126a4e)

will add screenshot for what this looks like after the change once I have my new laptop set up more.

lolbinarycat commented:
here's what it looks like after the change:

![screenshot](https://github.com/user-attachments/assets/7568dd22-7879-4485-bafd-1bea612b9ccd)

lolbinarycat commented:
This also increases consistency with mdBook, which has a much wider sidebar by default

![mdBook screenshot](https://github.com/user-attachments/assets/3ac2bba2-14f2-4c2d-b9b7-bb3b77f39040)

GuillaumeGomez commented:
The difference is (was?) that mdBook and even the rustdoc file panel are trees whereas the rustdoc sidebar is (was?) a list.

Anyway, I don't mind the change, we'll just do an FCP first to ensure that everyone else agrees with it. Once done, I can check what's wrong with the gui test failure.

@rfcbot fcp merge

bors commented:
:umbrella: The latest upstream changes (presumably #140895) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

camelid commented:
Is there a reason this is still marked as draft?

lolbinarycat committed:
rustdoc: rip out all the gui tests for <100% width mobile sidebar

stuff like "clicking the settings menu closes the mobile sidebar"
is now impossible for users to observe, since the mobile sidebar
will always cover the settings menu due to being full-width,
which is good because that behavior is also now impossible for
our testing framework to observe.

rust-log-analyzer commented:
The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/15196925260/job/42743236481) [(plain)](https://github.com/rust-lang/rust/commit/7c80c71a89df5c2ce69caa46125e2649a4a6c74e/checks/42743236481/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
info: removing rustup binaries
info: rustup is uninstalled
##[group]Image checksum input
mingw-check-tidy
# We use the ghcr base image because ghcr doesn't have a rate limit
# and the mingw-check-tidy job doesn't cache docker images in CI.
FROM ghcr.io/rust-lang/ubuntu:22.04

ARG DEBIAN_FRONTEND=noninteractive
RUN apt-get update && apt-get install -y --no-install-recommends \
  g++ \
---

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
#
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#12 2.772 Building wheels for collected packages: reuse
#12 2.773   Building wheel for reuse (pyproject.toml): started
#12 2.990   Building wheel for reuse (pyproject.toml): finished with status 'done'
#12 2.991   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132719 sha256=d2a2565e7037ad3883fb9337653f2e25bbb588534fbef3697286cbc26d1bf634
#12 2.991   Stored in directory: /tmp/pip-ephem-wheel-cache-67ldmsn9/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#12 2.994 Successfully built reuse
#12 2.994 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#12 3.401 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#12 3.402 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#12 3.962 Collecting virtualenv
#12 4.000   Downloading virtualenv-20.31.2-py3-none-any.whl (6.1 MB)
#12 4.095      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 6.1/6.1 MB 65.3 MB/s eta 0:00:00
#12 4.136 Collecting distlib<1,>=0.3.7
#12 4.140   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#12 4.147      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 86.3 MB/s eta 0:00:00
#12 4.185 Collecting platformdirs<5,>=3.9.1
#12 4.188   Downloading platformdirs-4.3.8-py3-none-any.whl (18 kB)
#12 4.238 Collecting filelock<4,>=3.12.2
#12 4.241   Downloading filelock-3.18.0-py3-none-any.whl (16 kB)
#12 4.323 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#12 4.520 Successfully installed distlib-0.3.9 filelock-3.18.0 platformdirs-4.3.8 virtualenv-20.31.2
#12 4.520 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#12 DONE 4.6s

#13 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#13 DONE 0.0s
---
DirectMap4k:      149440 kB
DirectMap2M:     7190528 kB
DirectMap1G:    11534336 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
##[group]Building bootstrap
    Finished `dev` profile [unoptimized] target(s) in 0.05s
##[endgroup]
WARN: currently no CI rustc builds have rustc debug assertions enabled. Please either set `rust.debug-assertions` to `false` if you want to use download CI rustc or set `rust.download-rustc` to `false`.
WARN: `rust.debug-assertions = true` will prevent downloading CI rustc as alt CI rustc is not currently built with debug assertions.
---
fmt check
fmt: checked 6022 files
tidy check
tidy: Skipping binary file check, read-only filesystem
##[error]tidy error: /checkout/tests/rustdoc-gui/sidebar-mobile.goml:76: tab character
##[error]tidy error: /checkout/tests/rustdoc-gui/sidebar-mobile.goml:77: tab character
removing old virtual environment
creating virtual environment at '/checkout/obj/build/venv' using 'python3.10' and 'venv'
creating virtual environment at '/checkout/obj/build/venv' using 'python3.10' and 'virtualenv'
Requirement already satisfied: pip in ./build/venv/lib/python3.10/site-packages (25.1.1)
linting python files
All checks passed!
checking python file formatting
26 files already formatted
checking C++ file formatting
some tidy checks failed
Command has failed. Rerun with -v to see more details.
Build completed unsuccessfully in 0:01:45
  local time: Thu May 22 21:22:36 UTC 2025
  network time: Thu, 22 May 2025 21:22:36 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

GuillaumeGomez commented on tests/rustdoc-gui/sidebar-mobile.goml:
```diff
@@ -72,6 +73,8 @@ define-function: (
+        press-key: "Escape"
     },
 )
 
```
If you could just check that the sidebar has the same width as the `<body>` (using [compare-elements-size](https://github.com/GuillaumeGomez/browser-UI-test/blob/master/goml-script.md#compare-elements-size), it'd be perfect.

GuillaumeGomez commented on tests/rustdoc-gui/sidebar-mobile.goml:
```diff
@@ -57,6 +57,7 @@ scroll-to: ".block.keyword li:nth-child(1)"
 compare-elements-position-near: (".block.keyword li:nth-child(1)", ".mobile-topbar", {"y": 544})
 
 // Now checking the background color of the sidebar.
+reload:
```
Why did you add this?

lolbinarycat replied:
it doesn't work without it (or a press-key: "Escape"), since we can no longer open the settings menu while the sidebar is open.

GuillaumeGomez replied:
Then instead use:

```
// Close the sidebar menu.
press-key: "Escape"
```

lolbinarycat committed:
rustdoc-gui test: apply suggestions from code review

lolbinarycat commented:
@GuillaumeGomez suggestions applied!

GuillaumeGomez commented:
Thanks!

@bors r+ rollup

bors commented:
:pushpin: Commit bcca611b359c74b71e798c2ee30f4695c6218bbf has been approved by `GuillaumeGomez`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=GuillaumeGomez bcca611b359c74b71e798c2ee30f4695c6218bbf -->
<!-- homu: {"type":"Approved","sha":"bcca611b359c74b71e798c2ee30f4695c6218bbf","approver":"GuillaumeGomez","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
